examples/99-advanced/atomic-orbitals.py

```diff
@@ -0,0 +1,226 @@
+    opac /= opac.max()
+    rgba[:, -1] = opac * 255
+
+    orbital['plot_scalars'] = rgba
+
+    pl = pv.Plotter()
+    vol = pl.add_volume(
+        orbital,
+        scalars='plot_scalars',
+    )
+    vol.prop.interpolation_type = 'linear'
+    if clip_plane:
+        pl.add_volume_clip_plane(
+            vol,
+            normal=clip_plane,
+            normal_rotation=False,
+        )
+    pl.camera_position = cpos
+    pl.camera.zoom(1.5)
+    pl.show_axes()
+    return pl.show()
+
+
+hydro_orbital = examples.load_hydrogen_orbital(3, 1, 0)
+plot_orbital(hydro_orbital, clip_plane='-x')
+
+
+###############################################################################
+# Volumetric Plot: 4dz2 orbital
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+hydro_orbital = examples.load_hydrogen_orbital(4, 2, 0)
+plot_orbital(hydro_orbital, clip_plane='-y')
+
+
+###############################################################################
+# Volumetric Plot: 4dxz orbital
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+hydro_orbital = examples.load_hydrogen_orbital(4, 2, -1)
+plot_orbital(hydro_orbital, clip_plane='-y')
+
+
+###############################################################################
+# Plot an Orbital Using a Density Plot
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We can also plot atomic orbitals using a 3D density plot. For this, we will
+# use :func:`numpy.random.choice` to sample all the points of our
+# :class:`pyvista.UniformGrid` based on the probability of the electron being
+# at that coordinate.
+#
+# Note how we square the probability to better visualize the distribution of
+# the atomic orbital.
+
+# Generate the orbital and sample based on the square of the probability of an
+# electron being within a particular volume of space.
+hydro_orbital = examples.load_hydrogen_orbital(4, 2, 0, zoom_fac=0.5)
+prob = np.abs(hydro_orbital['real_hwf']) ** 2
+prob /= prob.sum()
+indices = np.random.choice(hydro_orbital.n_points, 10000, p=prob)
+
+# add a small amount of noise to these coordinates to remove the "grid like"
+# structure present in the underlying UniformGrid
+points = hydro_orbital.points[indices]
+points += np.random.random(points.shape) - 0.5
+
+# Create a point cloud and add the phase of as the active scalars
+point_cloud = pv.PolyData(points)
+point_cloud['phase'] = hydro_orbital['real_hwf'][indices] < 0
+
+# Turn the point cloud into individual spheres. We do this so we can improve
+# the plot by enabling surface space ambient occlusion (SSAO)
+dplot = point_cloud.glyph(
+    geom=pv.Sphere(theta_resolution=8, phi_resolution=8), scale=False, orient=False
+)
+
+# be sure to enable SSAO here. This makes the "points" that are deeper within
+# the density plot darker.
+pl = pv.Plotter()
+pl.add_mesh(
+    dplot,
+    smooth_shading=True,
+    show_scalar_bar=False,
+    cmap=['red', 'green'],
+    ambient=0.2,
+)
+pl.enable_ssao(radius=10)
+pl.enable_anti_aliasing()
+pl.camera.zoom(2)
+pl.background_color = 'w'
+pl.show()
```

Comment:
I'm slightly curious how a "Gaussian points" representation of the sampled data would look, but I think the example is already extensive as it is (and the result might be uninteresting anyway).

Reply:
Good idea. Looks great!
![tmp](https://user-images.githubusercontent.com/11981631/215699477-09e83a9f-bdbf-4a3e-b4ab-02ad0bc2b797.png)